comptime.md

```diff
@@ -0,0 +1,125 @@
+# Compile-time execution
+
+# Summary 
+
+This RFC proposes to entirely change the current implementation of compile-time that will allow to run V language itself at comptime.
+
+# Motivation
+
+ The main purpose to do so is current limitation of compile-time evaluation:
+- No way to invoke arbitrary V functions
+- No way to choose function at compile time: 
+```v
+$if flag_a {
+    fn foo() {
+        println('bar')
+    }
+} $else {
+    fn foo() {
+        println('foo')
+    }
+}
+```
+^ Compiler cannot handle this code 
+
+- It limits our generics and compile-time reflection. With proper comptime it would be possible to generate complex serialization/deserialization functions without compiler support. 
+- Implementing proper comptime in fact would simplify our compiler pipelines, at the moment there is a lot of edge-cases for handling current limited comptime. 
+- Proper metaprogramming is must have in every modern language 
+
+
+# Guide-level explanation 
+
+With this RFC the main change is that all expressions and statements prefixed with `$` will be treated as compile-time and all of them can be executed *before* codegen. Current compile-time is quite limited and allows only a few statements (`$if`,`$for`) and only specific calls.
```

Comment:
Imho a distinct keyword like `comptime`, will be clearer than a sigil prefix like `$`, because it will be easier to read/skim without overlooking it, and to search for too, especially given that `$` is used in string interpolation.

Reply:
A separate keyword, will also make the transition period easier, since V could support the older `$if, $for` constructs too for a while, till `comptime` stabilizes.

Reply:
Yeah this is a good idea, I'll update RFC tomorrow and will also add better explanation to examples this evening or tomorrow :D